independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarResource.java

```diff
@@ -69,38 +52,34 @@ public void init() {
 
     @Override
     public byte[] getResourceData(String resource) {
-        final ZipFile zipFile = readLockAcquireAndGetJarReference();
-        try {
-            ZipEntry entry = zipFile.getEntry(resource);
+        return JarFileReference.withJarFile(jarFileReference, jarPath, resource, (jarFile, path, res) -> {
```

Comment:
Let's also remove the lambda please

Reply:
I introduced this lambda to generalize and centralize the management of the `jarFile` in the `JarFileReference` class as suggested by @franz1981. I took particular attention to make sure that both lambdas that I used for this call are non-capturing so I believe they shouldn't create any performance issue. At this point I see 3 possibilities:

1. Leave it as it is
2. Refactor the bodies of those lambdas into `private static` methods so we will be 100% sure that they are non-capturing
3. Revert the last commit and completely get rid of the lambdas (but bringing back some resposibilitis of the `jarFile` lifecycle into the `JarResource`)

My preference goes to option 2., but I'm not strongly opinionated on this, so @franz1981 @geoand please advice.

Reply:
Sorry, I should have been more clear: I meant replace the lambda with an anonymous class :)

Reply:
Ok, done.